### PR TITLE
Bug/dynamic table alter on column syntax

### LINKED ISSUE #769: [ADAP-885] [Bug] Dynamic table alter syntax and comment syntax is wrong
### Is this a new bug in dbt-snowflake?

- [X] I believe this is a new bug in dbt-snowflake
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

DBT tries to comment on the model with a wrong syntax statement 

`comment on dynamic_table model_name AS`

Instead of 

`comment on dynamic table model_name AS`

Also, during the alter process

it tries to do 

`alter dynamic_table sandbox.procs.customer alter `

instead of 

`alter dynamic table sandbox.procs.customer alter `

### Expected Behavior

`comment on dynamic table model_name AS`

and 

`alter dynamic table sandbox.procs.customer alter `

### Steps To Reproduce

To be able to change the 'dynamic_table' keyword to 'dynamic table'

### Relevant log output

_No response_

### Environment

```markdown
- OS:Mac os
- Python: 3.8.16
- dbt-core: 1.6.2
- dbt-snowflake: 1.6.2
```

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by kaarthik108**
resolves #
[docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose) dbt-labs/docs.getdbt.com/#

Resolves #769 

### Problem

Fix the alter syntax for dynamic table

### Solution

alter table table_name instead of alter dynamic table

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

**Comment by kaarthik108** [comment]
@mikealfare Just adding you here as you previously reviewed. Thanks

File: .changes/unreleased/Fixes-20230914-110800.yaml
Comment (mikealfare) [inline .changes/unreleased/Fixes-20230914-110800.yaml:1]: @kaarthik108 Can you please remove this changelog from this PR? It looks like this is the changelog from the previous PR, which is already in `main` under the 1.7 directory.

Comment (kaarthik108) [reply .changes/unreleased/Fixes-20230914-110800.yaml:1]: Done ✅

**Review by mikealfare — APPROVED** [review]

**Comment by mikealfare** [comment]
Thanks @kaarthik108 and @colin-rogers-dbt!

**Pull request merged**